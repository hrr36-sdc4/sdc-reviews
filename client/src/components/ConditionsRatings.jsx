import React from "react";

class ConditionsRatings extends React.Component {
  grabStars(rating, condition) {
    let users = this.props.reviews;
    // grab specific condtions rating from each user
    for (let i = 0; i < users.length; i++) {
      rating += users[i][condition];
    }
    // find the average rating from the users
    rating = Math.ceil(rating / users.length);

    // add the appropriate bold stars
    for (let i = 0; i < rating; i++) {
      var boldStar = document.createElement("span");
      boldStar.innerHTML = "&#9733;";
      var accurate = document.getElementsByClassName(condition);
      accurate[0].appendChild(boldStar);
    }

    // add the necessary unfilled stars
    if (rating != 5) {
      for (let i = rating; i < 5; i++) {
        var blankStar = document.createElement("span");
        blankStar.innerHTML = "&#9734;";
        var accurate = document.getElementsByClassName(condition);
        accurate[0].appendChild(blankStar);
      }
    }
  }

  render() {
    return (
      <div className="ConditionsRatings">
        <div className="ratings-container">
          <div>
            <div className="accuracy">
              Accuracy {this.grabStars(0, "accuracy")}
            </div>
            <div />
          </div>
          <div>
            <div className="communication">
              Communication {this.grabStars(0, "communication")}
            </div>
          </div>
          <div>
            <div className="cleanliness">
              Cleanliness {this.grabStars(0, "cleanliness")}
            </div>
          </div>
        </div>
        <div className="ratings-container">
          <div>
            <div className="location">
              Location {this.grabStars(0, "location")}
            </div>
          </div>
          <div>
            <div className="check_in">
              Check-In {this.grabStars(0, "check_in")}
            </div>
          </div>
          <div>
            <div className="value">Value {this.grabStars(0, "value")}</div>
          </div>
        </div>
      </div>
    );
  }
}

export default ConditionsRatings;
