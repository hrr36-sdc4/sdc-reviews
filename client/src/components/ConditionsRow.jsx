import React from "react";
import Col from "react-bootstrap/Col";
import Row from "react-bootstrap/Row";

class CondtionsRow extends React.Component {
  render() {
    const rating = this.props.rating || 0;

    const fullStars = data => {
      return [...Array(data)].map((x, i) => <span key={i}>&#9733;</span>);
    };

    const blankStars = data => {
      if (data !== 0) {
        return [...Array(data)].map((x, i) => <span key={i}>&#9734;</span>);
      }
    };

    return (
      <Row>
        <Col className={this.props.title}>
          <span>{this.props.title}</span>
        </Col>

        <Col>
          {" "}
          {fullStars(rating)}
          {blankStars(5 - rating)}
        </Col>
      </Row>
    );
  }
}

export default CondtionsRow;

{
  /* <Col className={this.props.title}>
        <span>{this.props.title}</span>
        {fullStars(rating)}
        {blankStars(5 - rating)}
      </Col> */
}
