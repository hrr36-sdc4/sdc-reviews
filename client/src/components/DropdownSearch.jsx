import React from "react";
import Col from "react-bootstrap/Col";
import DropdownButton from "react-bootstrap/DropdownButton";
import Dropdown from "react-bootstrap/Dropdown";

class DropDownSearch extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      value: "Most recent"
    };
  }

  change(e) {
    this.props.handleValueChange(e);
    e === "recent"
      ? this.setState({ value: "Most Recent" })
      : this.setState({ value: "Most Relevant" });
  }

  render() {
    return (
      <Col className="dropdown">
        <DropdownButton
          id="dropdown-basic-button"
          title={this.state.value}
          onSelect={this.change.bind(this)}
        >
          <Dropdown.Item eventKey="recent">Most Recent</Dropdown.Item>
          <Dropdown.Item eventKey="relevant">Most Relevant</Dropdown.Item>
        </DropdownButton>
      </Col>
    );
  }
}

export default DropDownSearch;
