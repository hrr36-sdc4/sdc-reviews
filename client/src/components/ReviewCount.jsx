import React from "react";
import Col from "react-bootstrap/Col";

class ReviewCount extends React.Component {
  render() {
    const average = this.props.average || 0;

    const fullStars = (data = 4) => {
      return [...Array(data)].map((x, i) => (
        // <span className="stars" key={i}>
        //   &#9733;
        // </span>
        <i class="fas fa-star stars"> </i>
      ));
    };

    const blankStars = data => {
      if (data !== 0) {
        return [...Array(data)].map((x, i) => (
          // <span key={i}>&#9734;</span>
          <i class="far fa-star stars" />
        ));
      }
    };

    return (
      <Col xs={12} md={12} lg={12} className="bottom-spacing">
        <h3 className="ReviewCount">
          {this.props.reviewLength} Reviews {fullStars(average)}
          {blankStars(5 - average)}
        </h3>
      </Col>
    );
  }
}

export default ReviewCount;
