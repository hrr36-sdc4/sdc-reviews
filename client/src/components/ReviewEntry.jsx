import React from "react";
import moment from "moment";
import Image from "react-bootstrap/Image";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

const ReviewEntry = props => (
  <li className="review-entry bottom-spacing">
    <Row className="description-spacing">
      <Col className="images" lg={2} sm={2} xs={2} xl={2}>
        <Image
          src={props.review.image_url}
          className="thumbnail"
          roundedCircle
        />
      </Col>
      <Col>
        <Row className="username">{props.review.username}</Row>
        <Row>
          {moment(props.review.created_at)
            .startOf("day")
            .fromNow()}
        </Row>
      </Col>
    </Row>
    <Row className="desciption-align">{props.review.description}</Row>
  </li>
);

export default ReviewEntry;
