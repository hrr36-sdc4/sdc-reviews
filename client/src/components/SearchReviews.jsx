import React from "react";

class SearchReviews extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      value: ""
    };
  }

  searchAction(event) {
    this.props.handleSearchInput(event.target.value);
    this.setState({
      value: event.target.value
    });
  }

  render() {
    return (
      <div>
        <div className="search-bar form-inline">
          <input
            className="form-control"
            type="text"
            value={this.state.value}
            onChange={this.searchAction.bind(this)}
          />
        </div>
        <div className="dropdown-menu">
          <select name="" id="">
            <option value="Most relevant">Most relevant</option>
            <option value="Most recent">Most recent</option>
          </select>
        </div>
      </div>
    );
  }
}

export default SearchReviews;
