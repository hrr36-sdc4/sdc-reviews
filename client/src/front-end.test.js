import { configure, mount, shallow, render } from "enzyme";
import React from "react";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
import toJson from "enzyme-to-json";
import renderer from "react-test-renderer";

global.shallow = shallow;
global.render = render;
global.mount = mount;

// Components to be tested
import ReviewCount from "./components/Reviewcount.jsx";
import SearchReviews from "./components/SearchReviews.jsx";
import Pagination from "./components/Pagination.jsx";
import ReviewEntry from "./components/ReviewEntry.jsx";
import ConditionsRatings from "./components/ConditionsRatings.jsx";
import DropdownSearch from "./components/DropdownSearch.jsx";
import ConditionsRow from "./components/ConditionsRow.jsx";

//  good for debugging -
// console.log(component.debug())
describe("tests ReviewCount Component", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ReviewCount debug />);

    expect(toJson(component)).toMatchSnapshot();
  });

  // reviewLength={this.state.reviews.length}
  //           average={ratings["totalAverage"]}
});

/////////  testing with MOUNT //////////////////////
// always call unmount() after to not affect other tests
////////////////////////////////////////////////////
/////////// wanting to test interacting with a child component then the mount method can be used.
describe("tests SearchReviews Component", () => {
  it("should render correctly", () => {
    const component = shallow(<SearchReviews />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it("should be possible to generate search critera with enter", () => {
    const component = mount(<SearchReviews />);
    component.find(".form-control").simulate("keydown", { keyCode: 13 });

    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it("should preventDefault when user presses enter on searching for reviews", () => {
    const value = "2";
    const onChange = jest.fn();
    const wrapper = shallow(<SearchReviews handleSearchInput={onChange} />);

    expect(wrapper).toMatchSnapshot();

    wrapper.find(".form-control").simulate("keydown", {
      keyCode: 13,
      preventDefault: () => {}
    });

    expect(wrapper).toMatchSnapshot();
  });
});
/////////// test fn calls
describe("tests Pagination Component", () => {
  const array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13];
  const mockFunction = jest.fn();

  // it("should render correctly", () => {
  //   // const component = renderer.create(
  //   //   <Pagination items={array} onChangePage={mockFunction} />
  //   // );
  //   // const component = mount(
  //   //   <Pagination items={array} onChangePage={mockFunction} />
  //   // );

  //   // expect(toJson(component)).toMatchSnapshot();
  //   // component.unmount();
  //   const tree = renderer
  //     .create(<Pagination items={array} onChangePage={mockFunction} />)
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  it("should call mockFunction on button click", () => {
    // const component = mount(<Pagination />);
    // expect(component.contains(<Pagination />)).toBe(true);

    const component = mount(
      <Pagination items={array} onChangePage={mockFunction} />
    );
    component.find("ul.pagination").simulate("click");
    expect(mockFunction).toHaveBeenCalled();

    component.unmount();
  });
});

/////////// test state updates
describe("tests ReviewEntry Component", () => {
  const review = {
    image_url: "image",
    username: "dawn",
    description:
      "Sodales mus cras fermentum hendrerit lorem. Sagittis hendrerit cubilia tincidunt nibh tortor posuere parturient quam consequat. Habitasse nec id massa nec tincidunt platea. Felis aliquam sed eget amet tortor quam. Condimentum ad. Ac condimentum, varius torquent posuere placerat faucibus fames etiam massa blandit platea ad eget venenatis nulla curabitur lectus turpis sociosqu imperdiet iaculis porttitor vitae tristique est parturient bibendum molestie non aenean semper montes ridiculus per. Mus. Dui leo ante nascetur aliquam urna quam, dignissim sit torquent ante. Facilisis. Metus. Pellentesque vel maecenas amet donec fringilla mus senectus. Curabitur suspendisse elit consectetuer placerat auctor etiam mattis sociosqu. Ridiculus ullamcorper felis sed hymenaeos Vitae, curabitur nunc metus platea aenean id ut vestibulum senectus. Feugiat hymenaeos nullam hymenaeos netus volutpat class aptent senectus est fusce urna torquent rutrum. Vel suspendisse."
  };
  const review_id = 1;

  it("should render correctly", () => {
    const component = shallow(<ReviewEntry key={review_id} review={review} />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it("sets loading state to true when clicking Read More button", () => {
    const component = mount(<ReviewEntry key={review_id} review={review} />);
    component.find("button").simulate("click");
    expect(component.state("expanded")).toEqual(true);
    component.unmount();
  });

  it("renders an image", () => {
    const wrapper = shallow(<ReviewEntry review={review} />);
    // const wrapper = component.find("img");
    expect(wrapper.prop("image_url")).toMatchSnapshot();
  });
});
//////////////////////////////////////////

// testing with RENDER
// renders to static html
// less costly than mount but less functionality

///////// testing with SHALLOW  //////////////////////
// renders only single component (not even its children)
// This is as with shallow(<MyComponent />), you're testing what MyComponent renders -
// not the element you passed into shallow
//////////////////////////////////////////
/////////// simple, non-interactive components
describe("tests ConditionsRatings Component", () => {
  const ratings = {
    accuracy: 1,
    communication: 1,
    cleanliness: 1,
    location: 1,
    check_in: 1,
    value: 1
  };

  ///////////// testing jsx
  test("accepts custom properties", () => {
    const wrapper = shallow(
      <ConditionsRatings ratings={ratings} flexBasis={0} flexGrow={1} />
    );
    expect(wrapper.prop("style")).toMatchSnapshot();
  });

  it("should render correctly", () => {
    const component = shallow(<ConditionsRatings ratings={ratings} />);

    expect(component).toMatchSnapshot();
  });
});

describe("tests DropdownSearch component", () => {
  const clickFn = jest.fn();

  it("should render correctly", () => {
    const component = shallow(<DropdownSearch handleValueChange={clickFn} />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it("selecting a dropdown option should trigger the props function", () => {
    const component = shallow(<DropdownSearch handleValueChange={clickFn} />);
    component.find("#dropdown-basic-button").simulate("change");
    expect(clickFn).toHaveBeenCalled();
  });
});

/////////// testing props
// test("should render title prop", () => {
//   const wrapper = shallow(<ConditionsRow title={"Title!"} rating={1} />);
//   expect(wrapper.prop("title")).toEqual("Title!");
// });
