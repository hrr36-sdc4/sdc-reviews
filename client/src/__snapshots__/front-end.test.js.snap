// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewCount should render correctly in "debug" mode 1`] = `
<ForwardRef(Bootstrap(Col))
  className="bottom-spacing"
  lg={12}
  md={12}
  xs={12}
>
  <h3
    className="ReviewCount"
  >
     Reviews 
    <i
      className="fas fa-star blank-star"
      key="0"
    />
    <i
      className="fas fa-star blank-star"
      key="1"
    />
    <i
      className="fas fa-star blank-star"
      key="2"
    />
    <i
      className="fas fa-star blank-star"
      key="3"
    />
    <i
      className="fas fa-star blank-star"
      key="4"
    />
  </h3>
</ForwardRef(Bootstrap(Col))>
`;

exports[`accept custom properties 1`] = `undefined`;

exports[`accepts custom properties 1`] = `undefined`;

exports[`pass a selected value to the onChange handler 1`] = `
<ForwardRef(Bootstrap(Col))
  className="form-group input-group"
  lg="3"
  md={true}
  sm={true}
  xs={true}
>
  <div
    className="input-group-prepend"
  >
    <span
      className="input-group-text"
    >
      <i
        className="fas fa-search"
      />
    </span>
  </div>
  <input
    className="form-control"
    onKeyPress={[Function]}
    placeholder="Search reviews"
    type="text"
  />
</ForwardRef(Bootstrap(Col))>
`;

exports[`pass a selected value to the onChange handler 2`] = `
<ForwardRef(Bootstrap(Col))
  className="form-group input-group"
  lg="3"
  md={true}
  sm={true}
  xs={true}
>
  <div
    className="input-group-prepend"
  >
    <span
      className="input-group-text"
    >
      <i
        className="fas fa-search"
      />
    </span>
  </div>
  <input
    className="form-control"
    onKeyPress={[Function]}
    placeholder="Search reviews"
    type="text"
  />
</ForwardRef(Bootstrap(Col))>
`;

exports[`should be possible to activate button with enter 1`] = `
<SearchReviews>
  <ForwardRef(Bootstrap(Col))
    className="form-group input-group"
    lg="3"
    md={true}
    sm={true}
    xs={true}
  >
    <Col
      as="div"
      bsPrefix="col"
      className="form-group input-group"
      lg="3"
      md={true}
      sm={true}
      xs={true}
    >
      <div
        className="form-group input-group col-lg-3 col-md col-sm col"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
          >
            <i
              className="fas fa-search"
            />
          </span>
        </div>
        <input
          className="form-control"
          onKeyPress={[Function]}
          placeholder="Search reviews"
          type="text"
        />
      </div>
    </Col>
  </ForwardRef(Bootstrap(Col))>
</SearchReviews>
`;

exports[`tests ConditionsRatings Component accepts custom properties 1`] = `undefined`;

exports[`tests ConditionsRatings Component should render correctly 1`] = `
<ForwardRef(Bootstrap(Container))
  className="ConditionsRatings"
>
  <ForwardRef(Bootstrap(Row))>
    <ForwardRef(Bootstrap(Col))
      lg={6}
      md={6}
      sm={12}
      xs={12}
    >
      <CondtionsRow
        rating={1}
        title="Accuracy"
      />
      <CondtionsRow
        rating={1}
        title="Communication"
      />
      <CondtionsRow
        rating={1}
        title="Cleanliness"
      />
    </ForwardRef(Bootstrap(Col))>
    <ForwardRef(Bootstrap(Col))
      lg={6}
      md={6}
      sm={12}
      xs={12}
    >
       
      <CondtionsRow
        rating={1}
        title="Location"
      />
      <CondtionsRow
        rating={1}
        title="Check In"
      />
      <CondtionsRow
        rating={1}
        title="Value"
      />
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(Row))>
</ForwardRef(Bootstrap(Container))>
`;

exports[`tests ConditionsRatings Component should render correctly with no props 1`] = `
<ForwardRef(Bootstrap(Container))
  className="ConditionsRatings"
>
  <ForwardRef(Bootstrap(Row))>
    <ForwardRef(Bootstrap(Col))
      lg={6}
      md={6}
      sm={12}
      xs={12}
    >
      <CondtionsRow
        rating={1}
        title="Accuracy"
      />
      <CondtionsRow
        rating={1}
        title="Communication"
      />
      <CondtionsRow
        rating={1}
        title="Cleanliness"
      />
    </ForwardRef(Bootstrap(Col))>
    <ForwardRef(Bootstrap(Col))
      lg={6}
      md={6}
      sm={12}
      xs={12}
    >
       
      <CondtionsRow
        rating={1}
        title="Location"
      />
      <CondtionsRow
        rating={1}
        title="Check In"
      />
      <CondtionsRow
        rating={1}
        title="Value"
      />
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(Row))>
</ForwardRef(Bootstrap(Container))>
`;

exports[`tests DropdownSearch component should render correctly 1`] = `
<ForwardRef(Bootstrap(Col))
  className="dropdown-column"
  lg="3"
  md={true}
  sm={true}
  xs={true}
>
  <DropdownButton
    id="dropdown-basic-button"
    onChange={[Function]}
    title="Most recent"
    variant="light"
  >
    <ForwardRef(ContextTransform)
      eventKey="recent"
    >
      Most recent
    </ForwardRef(ContextTransform)>
    <ForwardRef(ContextTransform)
      eventKey="relevant"
    >
      Most relevant
    </ForwardRef(ContextTransform)>
  </DropdownButton>
</ForwardRef(Bootstrap(Col))>
`;

exports[`tests Pagination Component should render correctly 1`] = `""`;

exports[`tests ReviewCount Component should render correctly in "debug" mode 1`] = `
<ForwardRef(Bootstrap(Col))
  className="bottom-spacing"
  lg={12}
  md={12}
  xs={12}
>
  <h3
    className="ReviewCount"
  >
     Reviews 
    <i
      className="fas fa-star blank-star"
      key="0"
    />
    <i
      className="fas fa-star blank-star"
      key="1"
    />
    <i
      className="fas fa-star blank-star"
      key="2"
    />
    <i
      className="fas fa-star blank-star"
      key="3"
    />
    <i
      className="fas fa-star blank-star"
      key="4"
    />
  </h3>
</ForwardRef(Bootstrap(Col))>
`;

exports[`tests ReviewEntry Component renders an image 1`] = `undefined`;

exports[`tests ReviewEntry Component should render correctly 1`] = `
<li
  className="review-entry bottom-spacing"
>
  <ForwardRef(Bootstrap(Row))
    className="description-spacing"
  >
    <ForwardRef(Bootstrap(Col))
      className="images"
      lg={3}
      sm={3}
      xl={3}
      xs={3}
    >
      <ForwardRef(Bootstrap(Image))
        className="thumbnail"
        roundedCircle={true}
        src="image"
      />
    </ForwardRef(Bootstrap(Col))>
    <ForwardRef(Bootstrap(Col))
      lg={6}
      sm={8}
      xl={6}
      xs={8}
    >
      <ForwardRef(Bootstrap(Row))
        className="username"
      >
        dawn
      </ForwardRef(Bootstrap(Row))>
      <ForwardRef(Bootstrap(Row))>
        19 hours ago
      </ForwardRef(Bootstrap(Row))>
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(Row))>
  <ForwardRef(Bootstrap(Row))
    className="desciption-align"
  >
    <ForwardRef(Bootstrap(Col))
      lg={12}
      sm={12}
      xl={12}
      xs={12}
    >
      Sodales mus cras fermentum hendrerit lorem. Sagittis hendrerit cubilia tincidunt nibh tortor posuere parturient
      <span>
        ...
        <button
          className="read-btn"
          onClick={[Function]}
        >
          Read More
        </button>
      </span>
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(Row))>
</li>
`;

exports[`tests ReviewEntry image renders and image 1`] = `undefined`;

exports[`tests SearchReviews Component should be possible to generate search critera with enter 1`] = `
<SearchReviews>
  <ForwardRef(Bootstrap(Col))
    className="form-group input-group"
    lg="3"
    md={true}
    sm={true}
    xs={true}
  >
    <Col
      as="div"
      bsPrefix="col"
      className="form-group input-group"
      lg="3"
      md={true}
      sm={true}
      xs={true}
    >
      <div
        className="form-group input-group col-lg-3 col-md col-sm col"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
          >
            <i
              className="fas fa-search"
            />
          </span>
        </div>
        <input
          className="form-control"
          onKeyPress={[Function]}
          placeholder="Search reviews"
          type="text"
        />
      </div>
    </Col>
  </ForwardRef(Bootstrap(Col))>
</SearchReviews>
`;

exports[`tests SearchReviews Component should preventDefault when user presses enter on searching for reviews 1`] = `
<ForwardRef(Bootstrap(Col))
  className="form-group input-group"
  lg="3"
  md={true}
  sm={true}
  xs={true}
>
  <div
    className="input-group-prepend"
  >
    <span
      className="input-group-text"
    >
      <i
        className="fas fa-search"
      />
    </span>
  </div>
  <input
    className="form-control"
    onKeyPress={[Function]}
    placeholder="Search reviews"
    type="text"
  />
</ForwardRef(Bootstrap(Col))>
`;

exports[`tests SearchReviews Component should preventDefault when user presses enter on searching for reviews 2`] = `
<ForwardRef(Bootstrap(Col))
  className="form-group input-group"
  lg="3"
  md={true}
  sm={true}
  xs={true}
>
  <div
    className="input-group-prepend"
  >
    <span
      className="input-group-text"
    >
      <i
        className="fas fa-search"
      />
    </span>
  </div>
  <input
    className="form-control"
    onKeyPress={[Function]}
    placeholder="Search reviews"
    type="text"
  />
</ForwardRef(Bootstrap(Col))>
`;

exports[`tests SearchReviews Component should render correctly 1`] = `
<ForwardRef(Bootstrap(Col))
  className="form-group input-group"
  lg="3"
  md={true}
  sm={true}
  xs={true}
>
  <div
    className="input-group-prepend"
  >
    <span
      className="input-group-text"
    >
      <i
        className="fas fa-search"
      />
    </span>
  </div>
  <input
    className="form-control"
    onKeyPress={[Function]}
    placeholder="Search reviews"
    type="text"
  />
</ForwardRef(Bootstrap(Col))>
`;
